commands/proposal_submission.go

```diff
@@ -1362,9 +1364,9 @@ func checkDataSourceSpec(spec *vegapb.DataSourceDefinition, name string, parentP
 
 				if ethOracle.Trigger != nil &&
 					ethOracle.Trigger.GetTimeTrigger() != nil &&
-					(ethOracle.Trigger.GetTimeTrigger().Initial == nil || *ethOracle.Trigger.GetTimeTrigger().Initial == 0) &&
-					(ethOracle.Trigger.GetTimeTrigger().Every == nil || *ethOracle.Trigger.GetTimeTrigger().Every == 0) {
-					errs.AddForProperty(fmt.Sprintf("%s.%s.external.ethoracle.trigger.timetrigger.(initial|every)", parentProperty, name), ErrIsRequired)
+					(ethOracle.Trigger.GetTimeTrigger().Initial == nil || *ethOracle.Trigger.GetTimeTrigger().Initial == 0) {
```

Comment:
I don't love that we're doing this in the command validation because I think its unexpected that a validation step would change values. For example this will be called by the CLI/Desktop wallet, so if I send in a proposal with `initial == nil` the transaction that will appear in the wallet to verify manually will have `initial` set to something and the transaction will look different than I asked it to send.

As I linked to you in the other PR, we do the exact same thing of setting `initial` to `enactmentTime` when its `nil` for internal time triggered oracles, and that happens in the governance engine: https://github.com/vegaprotocol/vega/blob/develop/core/governance/market.go#L583 We should just so that same here too.

Reply:
yes please, let's not change the transaction in this layer, this should be only validation, and the initial time if null should be applied in the governance engine. @ettec